### ISSUE TITLE
@interfaceObject is not supported.

### ISSUE THREAD

**Issue by wangmir** [99designs/gqlgen#2816]
### What happened?

`interfaceObject` is apollo federation directive. But, when I try to use it, it falsly generate directive handler at generated file.

it generates `InterfaceObject` at `DirectiveRoot` on generated `exec.go`

```go
type DirectiveRoot struct {
	ComposeDirective func(ctx context.Context, obj interface{}, next graphql.Resolver, name string) (res interface{}, err error)
	InterfaceObject  func(ctx context.Context, obj interface{}, next graphql.Resolver) (res interface{}, err error)
}
```

### What did you expect?

Seamlessly handling without adding new directive handler.

 I think it's because `interfaceObject` is missing from
https://github.com/99designs/gqlgen/blob/919aded6be6cd722c202214e7a8183ed56762501/plugin/federation/federation.go#L80-L88

### Minimal graphql.schema and models to reproduce

just adding `interfaceObject` on existing test data of federation2.

```graphql
extend schema
  @link(url: "https://specs.apollo.dev/federation/v2.3",
        import: ["@key", "@shareable", "@provides", "@external", "@tag", "@extends", "@override", "@inaccessible", "@interfaceObject"])

schema {
    query: CustomQuery
}

type Hello @key(fields:"name", resolvable: false) @interfaceObject {
  name: String!
}

type World @key(fields: "foo bar", resolvable: false) {
    foo: String!
    bar: Int!
}

extend type ExternalExtension @key(fields: "  upc    ") {
    upc: String!
    reviews: [Hello]
}

type CustomQuery {
  hello: Hello!
}
```

### versions
 - `go run github.com/99designs/gqlgen version`?
 0.17.38
 - `go version`?
 1.20.4

**Comment by wangmir** [comment]
It seems this PR (https://github.com/99designs/gqlgen/pull/2699) is the fix, but it's not getting in. @StevenACoffman can we add this?

**Comment by StevenACoffman** [comment]
The original author hasn't gotten it to pass both tests and linting and resolved the merge conflicts. Would you make a new PR that does?

**Comment by wangmir** [comment]
@StevenACoffman Ok, I'll try then.

**Comment by StevenACoffman** [comment]
Fixed in #2821
